dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -60,6 +61,8 @@
     private static final char URL_SEPARATOR = ' ';
     // URL address separated regular expression for parsing the service provider URL list in the file cache
     private static final String URL_SPLIT = "\\s+";
+    // Max times to retry to save properties to local cache file
+    private static final int MAX_RETRY_TIMES_SAVE_PROPERTIES = 3;

```

Comment:
I have some doubts about the retry scenario. In what scenario do we need to try again when the save file fails? Or what scenario has the first save failure, the second time may be successful?

Reply:
@ralf0131 
Would u pls also check this point? Thx
:)

Reply:
Er.. I'd say I am not quite sure why it is designed like this. One scenario I can think is that multiple process try to write to the same file, if other process is holding the file lock and this process might failed.  Does that sound reasonable?

Reply:
Hi,huxing
Thank you for your reply.

As far as I know, if multiple processes try to write data to the same file, `acquire FD` will be called at the bottom, and this operation is usually a lock operation, so I personally think that if this happens, actually will cause queuing instead of writing failure.

But I am not very sure about this point of view. We may need a local test to ensure the rigor of the logic? 
How do u think about it?
:)

Reply:
Yes, you can try it.

Reply:
Hi @carryxyh , do you have any conclusion? If not, I think I can have a try with it. :)